core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -511,17 +496,13 @@ oppia.directive('conversationSkin', [
 
           $scope.isCurrentCardAtEndOfTranscript = function() {
             return PlayerTranscriptService.isLastCard(
-              PlayerPositionService.getActiveCardIndex());
+              PlayerPositionService.getDisplayedCardIndex());
           };
-          var _addNewCard = function(
-              stateName, newParams, contentHtml, interactionHtml,
-              interaction) {
-            PlayerTranscriptService.addNewCard(
-              stateName, newParams, contentHtml, interactionHtml,
-              interaction, false);
-
-            if (newParams) {
-              LearnerParamsService.init(newParams);
+          var _addNewCard = function(newCard) {
+            PlayerTranscriptService.addNewCard(newCard);
+
+            if (newCard.getCurrentParams()) {
```

Comment:
"current params" is more like a player state thing.

Reply:
Done, now reinitializing params in submitAnswer() in the engine service.